src/map/lua/lua_baseentity.cpp

```diff
@@ -10766,6 +10785,62 @@ int32 CLuaBaseEntity::getAutomatonHead(lua_State* L)
     return 1;
 }
 
+int32 CLuaBaseEntity::isSpawned(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    return ((CMobEntity*)m_PBaseEntity)->PAI->IsSpawned();
+}
+
+int32 CLuaBaseEntity::isEngaged(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    return ((CMobEntity*)m_PBaseEntity)->PAI->IsEngaged();
+}
+
+int32 CLuaBaseEntity::clone(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_MOB);
+    DSP_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    CLuaBaseEntity* PClonee = Lunar<CLuaBaseEntity>::check(L, 1);
+    CStatusEffectContainer* statuses = ((CMobEntity*)PClonee->m_PBaseEntity)->StatusEffectContainer;
+    uint32 duration = 0;
+    EFFECT sleepEffects[3] = { EFFECT_SLEEP, EFFECT_SLEEP_II, EFFECT_LULLABY };
+
+    ((CMobEntity*)m_PBaseEntity)->health.hp = ((CMobEntity*)PClonee->m_PBaseEntity)->health.hp > 0 ? ((CMobEntity*)PClonee->m_PBaseEntity)->health.hp : ((CMobEntity*)m_PBaseEntity)->GetMaxHP();
+    for (int i = 0; i < 3; ++i)
+    {
+        EFFECT effect = sleepEffects[i];
+        if (statuses->HasStatusEffect(effect))
+        {
+            duration = statuses->GetStatusEffect(effect)->GetDuration();
+            ((CMobEntity*)m_PBaseEntity)->StatusEffectContainer->AddStatusEffect(new CStatusEffect(effect, effect, 1, 0, duration));
+            return 1;
+        }
+    }
+    
+    return 0;
+}
+
+int32 CLuaBaseEntity::isAsleep(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+
+    CStatusEffectContainer* statuses = ((CBattleEntity*)m_PBaseEntity)->StatusEffectContainer;
+    EFFECT sleepEffects[3] = { EFFECT_SLEEP, EFFECT_SLEEP_II, EFFECT_LULLABY };
```

Comment:
Since this is tangentially related, how do you go about making something sleep immune? I tried setting MOD_SLEEPRES to 100 but Elemental Seal still makes it land.